logic/LogicManager.java

```diff
@@ -69,7 +69,7 @@ public LogicManager(GraphManager graphMgr, ConceptManager conceptMgr, TraversalE
         this.compiledConjunctions = new HashMap<>();
     }
 
-    GraphManager graph() { return graphMgr; }
+    public GraphManager graph() { return graphMgr; }
```

Comment:
I think rather than getting the `GraphManager` from the `LogicManager` in the `CostEstimator` (and making this public), it would be better to pass it in as another argument to the constructor of the estimator.

One reason we have constructor arguments is to give a high-level signal of the main collaborators of the class. In your domain, answer estimation cannot be done without access to the graph, so we should highlight that by passing in the constructor explicitly.